File: app/src/main/java/org/scp/gymlog/room/Connection.java
package org.scp.gymlog.room;

import android.content.Context;

import androidx.room.Room;

public class Connection {

    private static AppDatabase db;

    public static AppDatabase get(Context context) {
        if (db == null) {
            db = Room.databaseBuilder(context, //getApplicationContext(),
                    AppDatabase.class, "gymlog-db").build();
        }
        return db;
    }
}

File: app/src/main/java/org/scp/gymlog/room/Connection.kt
package org.scp.gymlog.room

import android.content.Context
import androidx.room.Room

object Connection {

    private var db: AppDatabase? = null

    operator fun get(context: Context): AppDatabase {
        if (db == null) {
            db = Room.databaseBuilder(context,  //getApplicationContext(),
                AppDatabase::class.java, "gymlog-db").build()
        }
        return db!!
    }
}
